src/app/components/analytics/host-analytics/host-analytics.component.html

```diff
@@ -0,0 +1,117 @@
+              <img *ngIf="currentChallengeDetails['image']!==null" [src]='currentChallengeDetails["image"]'>
+              <img *ngIf="currentChallengeDetails['image'] == null" src='assets/images/404.png'>
+            </div>
+          </div>
+          <div class="col m8 l10">
+            <div class="card-content analytics-card-content ">
+              <h3 class="fs-20 rm-mt"><span>{{currentChallengeDetails['title']}}</span>
+              </h3>
+              <span class="text-light-black">Hosted by: </span> {{currentChallengeDetails['creator']['team_name']}}
+            </div>
+          </div>
+        </div>
+        <div class="analytics-challenge-card">
+          <div class="main-title fs-12">
+            Short Description
+          </div>
+          <p class="no-top-para">
+            {{currentChallengeDetails['short_description']}}
+          </p>
+        </div>
+        <div class="analytics-challenge-card fs-12 analytics-challenge-single-line">
+                    <span class="main-title">
+                        Total Participant Teams:
+                    </span>
+          <span class="show-count">
+                        {{totalChallengeTeams}}
+                    </span>
+          <a class="participant-download-btn waves-effect waves-dark btn ev-btn-dark fw-light fs-12" (click)="downloadChallengeParticipantTeams()">Download</a>
+        </div>
+        <div class="analytics-challenge-card">
+          <div class="main-title fs-12">
+            Challenge Phases
+          </div>
+          <div class="analytics-phases-card" *ngFor="let item of currentPhase">
+            <p class="no-top-para">
+              {{item.name}}
+            </p>
+            <div class="text-light-black fs-12">
+              Challenge Phase Duration
+            </div>
+            <p class="no-top-para">
+              {{item.start_date | date:'medium'}} - {{item.end_date | date:'medium'}}
+            </p>
+            <div class="row center">
+              <div class="col m3">
+                <div class="text-light-black fs-12">
+                  Total Submissions
+                </div>
+                <p class="no-top-para fs-20">
+                  {{totalSubmission[item.id] || 0}}
+                </p>
+                <div class="short-hr-line"></div>
+              </div>
+              <div class="col m3">
+                <div class="text-light-black fs-12">
+                  Participant Teams
+                </div>
+                <p class="no-top-para fs-20">
+                  {{totalParticipatedTeams[item.id] || 0}}
+                </p>
+                <div class="short-hr-line"></div>
+              </div>
+              <div class="col m3">
+                <div class="text-light-black fs-12">
+                  Latest Submission At
+                </div>
+                <p *ngIf="lastSubmissionTime[item.id]" class="no-top-para mb-5">
+                  {{lastSubmissionTime[item.id] | date:'medium'}}
```

Comment:
Here, when you have no submissions, `lastSubmissionTime[item.id]` is returned as _**You dont have any submissions in this challenge phase!**_. So, eventually, it enters under the `*ngIf` but then leads to a console error as that string is incompatible with the `Date` pipe.

Reply:
yeah, I think I have handled that case in the logic. i.e **lastsubmissionTime[item.id]** will either be **date string** or **undefined**

Reply:
are you getting error on your machine regarding that?

Reply:
resolved?

Reply:
Can you try when no. of submissions in a phase are zero? Because I seem to be getting the above error.

Reply:
I have tried that for zero participants also. it shows none as expected

Reply:
This, when I again tried in a different machine.

![pipe_error](https://user-images.githubusercontent.com/21227893/61628956-3aadba00-acb6-11e9-884e-14bb23ec4f0f.png)

Reply:
Thanks for resolving this. The error is gone now.